models/user.go

```diff
@@ -104,20 +94,45 @@ type Consens struct {
 	Mail            string         `json:"-" firestore:"-" bigquery:"-"`
 }
 
-type Address struct {
-	StreetName   string `json:"streetName" firestore:"streetName" bigquery:"-"`
-	StreetNumber string `json:"streetNumber" firestore:"streetNumber" bigquery:"-"`
-	City         string `json:"city" firestore:"city" bigquery:"-"`
-	PostalCode   string `json:"postalCode" firestore:"postalCode" bigquery:"-"`
-	Locality     string `json:"locality" firestore:"locality" bigquery:"-"`
-	CityCode     string `json:"cityCode" firestore:"cityCode" bigquery:"-"`
-	Area         string `json:"area" firestore:"area" bigquery:"-"`
+func UnmarshalUser(data []byte) (Claim, error) {
+	var r Claim
+	err := json.Unmarshal(data, &r)
+	return r, err
+}
+
+func (u *User) Marshal() ([]byte, error) {
+	return json.Marshal(u)
+}
+
+func (u *User) Sanitize() {
+	u.Name = lib.ToUpper(u.Name)
+	u.Surname = lib.ToUpper(u.Surname)
+	u.Gender = lib.ToUpper(u.Gender)
+	u.FiscalCode = lib.ToUpper(u.FiscalCode)
+	u.VatCode = lib.ToUpper(u.VatCode)
+	u.Mail = lib.ToUpper(u.Mail)
+	u.Phone = lib.TrimSpace(u.Phone)
+	if u.Residence != nil {
+		u.Residence.Sanitize()
+	}
+	if u.Domicile != nil {
+		u.Domicile.Sanitize()
+	}
+	u.BirthDate = lib.TrimSpace(u.BirthDate)
+	u.BirthCity = lib.ToUpper(u.BirthCity)
+	u.BirthProvince = lib.ToUpper(u.BirthProvince)
+	u.Work = lib.TrimSpace(u.Work)
+	u.WorkType = lib.TrimSpace(u.WorkType)
+	u.WorkStatus = lib.TrimSpace(u.WorkStatus)
+	for index, _ := range u.IdentityDocuments {
+		u.IdentityDocuments[index].Sanitize()
+	}
```

Comment:
IdentityDocument is saved as a pointer... could this explode?
should we make something like this?
if key, ok := u.IdentityDocuments[index]; ok {
      key.Sanitize()
}

Reply:
I didn't put any check because IdentityDocuments if a field of type `[]*IdentityDocument` so I've assumed that it won't happend to have a non empty array with an IdentityDocument nil